### PR TITLE
resample jit

### PR DESCRIPTION
**Pull request by vincentqb**
Thanks to @Oktai15 in #362 for bringing up support for jit and `resample_waveform`.

### CODE REVIEW

File: torchaudio/compliance/kaldi.py [hunk 1]
```diff
@@ -20,7 +19,7 @@
 ]
 
 # numeric_limits<float>::epsilon() 1.1920928955078125e-07
-EPSILON = torch.tensor(torch.finfo(torch.float).eps, dtype=torch.get_default_dtype())
+EPSILON = torch.tensor(torch.finfo(torch.float).eps)
```
Comment (Oktai15) [inline torchaudio/compliance/kaldi.py]: It will create EPSILON always on CPU, it is not right?

File: torchaudio/compliance/kaldi.py [hunk 2]
```diff
@@ -92,10 +107,10 @@ def _feature_window_function(window_type, window_size, blackman_coeff):
         # like hanning but goes to zero at edges
         return torch.hann_window(window_size, periodic=False).pow(0.85)
     elif window_type == RECTANGULAR:
-        return torch.ones(window_size, dtype=torch.get_default_dtype())
+        return torch.ones(window_size)
```
Comment (Oktai15) [inline torchaudio/compliance/kaldi.py]: The same question: CPU tensor?

**Comment by Oktai15** [comment]
By the way, you can try to do jit kaldi.fbank in this PR (it needs very small changes I suppose)

File: torchaudio/compliance/kaldi.py (hunk 1 above)
Comment (vincentqb) [reply torchaudio/compliance/kaldi.py]: This code needs updating to work on GPU, and we don't officially support GPU for now in torchaudio. I won't do this as part of this PR.

File: torchaudio/compliance/kaldi.py [hunk 3]
```diff
@@ -92,10 +107,10 @@ def _feature_window_function(window_type, window_size, blackman_coeff):
         # like hanning but goes to zero at edges
         return torch.hann_window(window_size, periodic=False).pow(0.85)
     elif window_type == RECTANGULAR:
-        return torch.ones(window_size, dtype=torch.get_default_dtype())
```
Comment (zhangguanheng66) [inline torchaudio/compliance/kaldi.py]: why don't we force the torch default type?

Comment (vincentqb) [reply torchaudio/compliance/kaldi.py]: torchscript does not support calling `torch.get_default_dtype`. In any case, the `dtype` of `torch.tensor`, `torch.ones` and `torch.arange` is already set to `torch.get_default_dtype`, see [torch.tensor](https://pytorch.org/docs/stable/torch.html#torch.tensor), [torch.ones](https://pytorch.org/docs/stable/torch.html#torch.ones) and [torch.arange](https://pytorch.org/docs/stable/torch.html#torch.arange). Is that what you meant?

Comment (vincentqb) [reply torchaudio/compliance/kaldi.py]: (relates to pytorch/pytorch#30774)

**Comment by vincentqb** [comment]
Pushed to #362.

**Pull request closed without merging**